### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by TheCloneMaster**
Updated models, completed wizards, added email confirmations with templates and several code fixes.
Also fixes issues  
   #17 [design] Patient should be res.partner not user.
   #16 [ERROR] patient active=False by default.

### CODE REVIEW

File: medical/__init__.py [hunk 1]
```diff
@@ -22,5 +22,6 @@
 
 import medical_constants
```
Comment (pedrobaeza) [inline medical/__init__.py]: Use relative import `from . import ...`

File: medical/views/medical_appointment_view.xml [hunk 2]
```diff
@@ -28,15 +28,9 @@
...
-                        <button name="button_confirm" states="draft" string="Confirm" class="oe_highlight"  type="object" groups="base.group_user"/>
-                        <button name="button_waiting" states="confirm" string="Waiting" class="oe_highlight"  type="object" groups="base.group_user"/>
-                        <button name="button_in_consultation" states="waiting" string="In Consultation" class="oe_highlight"  type="object" groups="base.group_user"/>
-                        <button name="button_done" states="in_consultation" string="Done" class="oe_highlight"  type="object" groups="base.group_user"/>
-                        <button name="button_cancel" states="confirm,in_consultation" string="Cancel"  type="object" groups="base.group_user"/>
-                        <field name="state" widget="statusbar" statusbar_visible="draft,confirm,waiting,in_consultation,done,canceled" statusbar_colors='{"canceled":"red","waiting":"blue"}'/>
+	                    <field name="stage_id" widget="statusbar" clickable="True"
```
Comment (pedrobaeza) [inline medical/views/medical_appointment_view.xml]: Don't use tabs, but spaces

**Comment by pedrobaeza** [comment]
Check also PEP8 issues.

File: medical/models/medical_appointment.py [hunk 3]
```diff
@@ -21,16 +21,80 @@
 # #############################################################################
 
 import time
-from datetime import datetime
+from datetime import datetime, date, timedelta
+import pytz
 
 from openerp.osv import fields, orm
+from openerp.tools.translate import _
```
Comment (yvaucher) [inline medical/models/medical_appointment.py:27]: You can import `_` directly from openerp

```
from openerp import _
```

File: medical/__init__.py (hunk 1 above)
Comment (TheCloneMaster) [inline medical/__init__.py]: done

File: medical/views/medical_appointment_view.xml (hunk 2 above)
Comment (TheCloneMaster) [inline medical/views/medical_appointment_view.xml]: done

**Comment by TheCloneMaster** [comment]
I updated files for line length <= 120, is it requisite to have them <80 ?

Also, I don't know how to fix yaml test error....  I tried adding "medical." to oe_specialty_1 and without but the error is still triggered...   Please help with that....

**Comment by pedrobaeza** [comment]
The length should be <80

**Comment by yvaucher** [comment]
@TheCloneMaster You should definitively have a look at what I did in https://github.com/OCA/vertical-medical/pull/6

For exemple, for some readibility you could use https://github.com/OCA/vertical-medical/pull/6/files#diff-3342df827adae2a46375586cb86b0046R29

I wonder if you could have rebase on #6 by the way it's a shame to do twice what was already done.

**Comment by TheCloneMaster** [comment]
PEP8 cleanup completed.

Please help me fix the failing yaml test...

File: medical_emr/models/medical_prescription_line.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        result = {}
+
+        # if name == 'form':
+        # result = {'value': {
+        # 'form' : medication_obj.browse(
+        #     cr, uid, medication, context = None).form.id ,
+        # } }
+        return result
+
+    # def _get_dose(self, cr, uid, ids, field_name, arg, context=None):
```
Comment (nhomar) [inline medical_emr/models/medical_prescription_line.py:41]: @TheCloneMaster can you delete this commented code.?

File: medical_emr/models/medical_prescription_order.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    def print_prescription(self, cr, uid, ids, context=None):
+        '''
+        '''
+        #        assert len(ids) == 1, 'This option should only be used for '
+        #                              'a single id at a time'
+        #        wf_service = netsvc.LocalService("workflow")
+        #        wf_service.trg_validate(uid, 'medical.prescription.order',
+        #                                ids[0], 'prescription_sent', cr)
```
Comment (nhomar) [inline medical_emr/models/medical_prescription_order.py:66]: @TheCloneMaster commented code not necesary.

**Comment by yvaucher** [comment]
@TheCloneMaster 

```
 ValueError: ref('oe_specialty_1') not found when processing /home/travis/build/OCA/vertical-medical/medical/tests/physician.yml.
This Yaml file appears to depend on missing data. This often happens for tests that belong to a module's test suite and depend on each other.
```

I think you removed or renamed `oe_specialty_1` otherwise it could be that this demo data is in another module so you need to rename it.

**Comment by TheCloneMaster** [comment]
Hi,

Nhomar, commented code is from previous version and should be fixed and put back, that's why I didn't remove it.

I think everything else is clear but the yaml test.
I didn't create these tests and I have tried to fix them but they keeps failing...  'oe_specialty_1' is defined right before the error message is triggered, so I don't know what is going on.  Pleas help !

Regards,

**Comment by yvaucher** [comment]
@TheCloneMaster have you tried to restore that `medical.` in `ref()` ?

**Comment by TheCloneMaster** [comment]
@yvaucher It was failing originally with medical. in ref(), but will try again...
Updating PR right now...

Thanks!

**Comment by TheCloneMaster** [comment]
Hi, seems like CI is having a hard time...

  File "/home/travis/OCB-8.0/openerp/models.py", line 48, in <module>
    import pytz
ImportError: No module named pytz

**Comment by TheCloneMaster** [comment]
I give up, no way to get that yml test right...   :-(

Tried all the ways I could find in current samples, all of them triggered error...

I checked previous PRs and that yml test was failing since the begining...How can we move forward ?

**Comment by ajinkyagadewar** [comment]
Hello TheCloneMaster,

What is the update? Is vertical-medical ready for Odoo 8.0?

**Comment by yvaucher** [comment]
@TheCloneMaster Is this test working locally?

**Comment by lasley** [comment]
@TheCloneMaster - I merged changes from this branch into mine in #20, and fixed the slew of test issues. Is there a list of features you were looking to add in this ticket, or was everything done just pending the test fixes?

**Comment by TheCloneMaster** [comment]
Hi Dave,

Great !!

So now should I remove my PR?  Or is it fine for maintainers to merge it
and then merge yours ?

Next thing is to fix field placement for medical_emr views and re-introduce
python logic, as most forms right now are totally "dumb".

Then, fix oemedical_invoice and move it to installable modules...

Would you like to take on any of these?  I will take the other one...

Regards,

**Comment by lasley** [comment]
Hi @TheCloneMaster - You can just go ahead and close this one, everything will be included when my PR is merged.

I can work on porting oemedical_invoice (and likely oemedical_insurance at the same time). Just to note, there seemed to be inconsistent usage of "medical"/"Medical" vs "oemedical"/"OeMedical". I went ahead and removed the prefix from everything, it looks cleaner IMO :wink:

**Comment by TheCloneMaster** [comment]
All changes have been included in PR #20, so I am closing this one...

**Comment by TheCloneMaster** [comment]
Done !

Then I will wait for merge of your branch and then will continue with
fields placement on medical_emr views...

Regards,

**Comment by lmignon** [comment]
@dlasley Thank you for your work to finalize the migration started by @TheCloneMaster. @TheCloneMaster Thank you for your initial work and the fixes done previously to #20. (especially #17 [design] Patient should be res.partner not user.)

**Comment by TheCloneMaster** [comment]
Community work my friend...   ;-)

We all benefit from this...

**Pull request closed without merging**